### ISSUE TITLE
Some page sections are no longer translated

### ISSUE THREAD

**Issue by jmaspons** [r-lib/pkgdown#2832]
For example, https://osm-catalan.github.io/monitorOSM/docs/ should be translated in Catalan as per https://github.com/OSM-Catalan/monitorOSM/blob/af9ec15532fab3a0c23afb56022e9c7e1dc021d8/_pkgdown.yml#L4. It used to work, but not with pkgdown 2.1.1. I don't know when it starts failing.

**Comment by jayhesselberth** [comment]
The site you links appears to be in Catalan (at least my browser prompts me to convert Catalan to English when I visit), it's built with 2.1.1 (version number at the bottom of the page). Not sure what you're seeing?

**Comment by jayhesselberth** [comment]
I added `lang: es` to pkgdown and rebuilt the home page. Everything *except* the main section is translated (the opposite of what you're seeing). Weird.

<img width="1313" alt="image" src="https://github.com/user-attachments/assets/1c804168-a939-4053-945a-d5011b3e371e" />

**Comment by hadley** [comment]
@jayhesselberth that's because the README is already in Catalan: https://github.com/OSM-Catalan/monitorOSM/?tab=readme-ov-file#monitorosm-

Do you get a translated navbar if you build OSM-Catalan docs locally?

**Comment by jmaspons** [comment]
> Do you get a translated navbar if you build OSM-Catalan docs locally?

Yes, it works locally! So some problem with github actions I guess. Feel free to close the issue if you think there is nothing wrong here

**Comment by ror-at-ebp** [comment]
I ran into the same issue with my package [mvwizr](https://ror-at-ebp.github.io/mvwizr/). To this end I created a minimal package to try to reproduce the issue - in the `_pkgdown.yml` config of the [langtest](https://github.com/ror-at-ebp/langtest) Package, the parameter `lang: de` is set.

As expected, the site is translated to German when rendered locally using `pkgdown::build_site` but not when rendered using the pkgdown github workflow.

I have also tried reproducing the issue locally with [Act](https://github.com/nektos/act) (on macOS ARM64 and Linux AMD64), but that resulted in the correct translated website too.

Unrelated: There is a typo in the German translation for developers. It says "Entwicker:innen" instead of "Entwickler:innen". #

**Comment by jayhesselberth** [comment]
Looking through the sites linked in this issue, it seems this problem still persists. But IIUC this is only a problem when building through Actions. Is it possible we need to update the Actions runner in some way to properly translate sites?

**Comment by hadley** [comment]
@jayhesselberth ooooh maybe. You might need to install a language package on the actions server.

**Comment by ror-at-ebp** [comment]
Just a remark - not sure if this should have any effect: I had already modified the workflow document to generate the German-Swiss locale before building the site. This fixed another issue (not being able to use that locale in ggplot, e.g., for month labels), but didn't fix the issue with the missing translation of the pkgdown website [(modified github action workflow):](https://github.com/ror-at-ebp/langtest/blob/67d540ce3d21883ab9f89ba60eaf49abc88ead5e/.github/workflows/pkgdown.yaml)

```
      - name: Set locale to German
        run: |
          sudo apt-get update
          sudo apt-get install -y locales
          sudo locale-gen de_CH.UTF-8
          sudo update-locale LANG=de_CH.UTF-8
        shell: bash

      - name: Build site
        run: |
          Sys.setlocale("LC_ALL", "de_CH.UTF-8") 
          pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)
        shell: Rscript {0}
```

**Comment by jayhesselberth** [comment]
I got this to work by updating the locale *and* setting the `env` during build:

https://github.com/jayhesselberth/pkgdown-gha-language/blob/e7989f869667d57e686e2efb65d9eb1de82b3a9f/.github/workflows/pkgdown.yaml#L39-L52
